platforms/core-runtime/build-configuration/src/main/java/org/gradle/internal/buildconfiguration/tasks/UpdateDaemonJvmTask.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.buildconfiguration.tasks;
```

Comment:
This is a good point, however, potentially I'll consider moving this when necessary during the implementation of the DSL on `Phase 3`, until then probably will be great to keep it as internal similar to the existing `javaToolchains` task.

However, happy to change this if we consider this a must

Reply:
Not necessarily a must right now but in the long run I think this will need to be public. 

We will be generating accessors with this type, and they should reference a non-internal type.